ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlVersion.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class KsqlVersion {
+
+  private static final Pattern VERSION_PATTERN = Pattern.compile("v?([0-9]+)\\.([0-9]+)\\..*");
+  private static final Map<KsqlVersion, KsqlVersion> versionMapping = new HashMap<>();
+
+  private static final KsqlVersion CP_60 = new KsqlVersion("6.0.");
+  private static final KsqlVersion CP_61 = new KsqlVersion("6.1.");
+
+  private static final KsqlVersion V_010 = new KsqlVersion("0.10.");
+  private static final KsqlVersion V_014 = new KsqlVersion("0.14.");
+
+  static {
+    versionMapping.put(CP_60, V_010);
+    versionMapping.put(CP_61, V_014);
+  }
+
+  private enum VersionType {
+    CONFLUENT_PLATFORM,
+    KSQLDB_STANDALONE
+  }
+
+  private final VersionType versionType;
+  private final String originalVersion;
+  private final int majorVersion;
+  private final int minorVersion;
+
+
+
+  public KsqlVersion(final String version) {
+    final Matcher matcher = VERSION_PATTERN.matcher(version);
+    if (!matcher.find()) {
+      throw new IllegalArgumentException("Unexpected ksqlDB server version: " + version);
+    }
+
+    originalVersion = version;
+    majorVersion = Integer.parseInt(matcher.group(1));
+    minorVersion = Integer.parseInt(matcher.group(2));
+    versionType = majorVersion < 4
+        ? VersionType.KSQLDB_STANDALONE
+        : VersionType.CONFLUENT_PLATFORM;
+  }
+
+  public boolean isAtLeast(final KsqlVersion version) {
+    if (versionType == version.versionType) {
+      return isAtLeastVersion(this, version.majorVersion, version.minorVersion);
+    }
+
+    if (versionType == VersionType.KSQLDB_STANDALONE) {
+      final KsqlVersion otherStandalone = versionMapping.get(version);
+      return otherStandalone != null
```

Comment:
Looks like this method returns false if the two versions are not comparable (because one is a standalone version and one is a CP version). Can we add this as a javadoc so future readers aren't left wondering?